Bottom: fd1d725d2676eb9abe694b4034a8afc2edd36081
Top:    2fab5dd88e216f1c46e2e38d7a17d74218db5b12
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-09-09 12:59:48 +0100

Fix layer on/off rendering

I'm not certain this is correct at all. It works around failure
to switch layer visibility on / off after

commit 9768e060fad7bc3dfc366da76ea1db8154005018
"Fix layer visibility logic for some boards."

After which, copper layers on the component / solder group
would still be drawn unless the silk layer was also turned off.

This appears to be due to the GL HID's custom drawing routines
not checking the return of the (HID)->set_layer function in certain
instances.

This fix is almost certainly WRONG.


---

diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index ecc02ff..bf0c723 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -1540,7 +1540,7 @@ ghid_draw_everything (BoxTypePtr drawn_area)
     l = global_view_2d ? LAYER_ON_STACK (i) : LAYER_PTR (orderi);
     group = GetLayerGroupNumberByNumber (global_view_2d ? LayerStack[i] : orderi);
 
-    if (/*l->On && */!do_group[group]) {
+    if (l->On && !do_group[group]) {
       do_group[group] = 1;
       drawn_groups[ngroups++] = group;
     }
